oop-relationships/oop-relationships.md

```diff
@@ -0,0 +1,132 @@
+# OOP Relationships
+
+## Learning Goals
+
+- Define Inheritance in OOP as an "is a" relationship: superclass, subclass
+- Define Composition in OOP as a "has a" relationship
+
+## Introduction
+
+When multiple objects interact with each other, they form a relationship. Like any relationship, two classes having a relationship should spark many questions for us: What do the classes do together? How do they communicate with each other? What's the nature of their relationship? Is one class in charge? Are the two classes changing and modifying each other, or is it one-way? What happens if one class changes, and the other stays stagnant?
+
+Because OOP is a paradigm that encourages forming relationships between these objects, OOP principles describe several types of these relationships. Two of these types of relationships are _inheritance_ and _composition_.
+
+These relationships will affect _how_ we solve problems that involve more than one class. Eventually, these types of relationships get applied when following software design patterns.
+
+## Vocabulary and Synonyms
+
+| Vocab | Definition | Synonyms | How to Use in a Sentence
+| --- | --- | --- | ---
+
+## Inheritance: Passing on Implementation
+
+Sometimes, two classes are similar... but different. The two classes share the same attributes and the same methods... but one class has additional attributes, additional methods, and a different name. We could formalize a relationship between these two classes with inheritance.
+
+Inheritance designates a **parent class** and a **child class**. Using inheritance, the _child class_ will "copy" the state and behavior defined in the _parent class_.
+
+When a _child class_ tries to use an attribute or method that it doesn't define in its own class, it will look up to the _parent class_. If the parent class has that attribute or method defined, the child class will use that implementation.
```

Comment:
```suggestion
Imagine reading an attribute or calling a method from an instance of a child class.

If Python does not see a definition for that attribute or method in the _child class_, Python will try looking for it in the _parent class_. If the parent class has that attribute or method defined, the child class will be able to use that implementation.
```

What do we think about blowing up the structure of the first sentence?

I'm not perfect at reworking it, but here's a stab. 🗡️ 

I'm a little troubled because inheritance isn't reserved for just instances as my suggestion might imply, but also words, and examples, they are hard 🤷

Reply:
Yes yes yes.  This even hints at the runtime lookup.  Thank you!